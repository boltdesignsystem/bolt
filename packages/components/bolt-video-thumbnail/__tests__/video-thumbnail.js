import { render, stopServer, renderWC } from '../../../testing/testing-helpers';
import schema from '../video-thumbnail.schema';
// eslint-disable-next-line camelcase
const { border_radius, aspect_ratio } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: '/fixtures/1200x660.jpg',
  };

  const chipObject = `{% set chip %}
    {% include '@bolt-components-chip/chip.twig' with {
      text: 'Preview',
      size: 'small',
      border_radius: 'small',
      color: 'violet',
    } only %}
  {% endset %}`;

  const videoData = {
    video: {
      title: 'Title of the Video',
      duration: '4:55',
    },
  };

  const videoObject = `<video-js
    data-account='1900410236'
    data-player='O3FkeBiaDz'
    data-embed='default'
    data-video-id='4892122320001'
    controls
    class="c-base-video"></video-js>`;

  fixtures = {
    defaultData,
    chipObject,
    videoData,
    videoObject,
  };
});

describe('Bolt Video Thumbnail Component', () => {
  test(`default`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
      },
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt Video Thumbnail Component Prop -', () => {
  // // Target each of the schema keys with the following pattern
  border_radius.enum.forEach(async option => {
    test(`border_radius items: ${option}`, async () => {
      const results = await render(
        '@bolt-components-video-thumbnail/video-thumbnail.twig',
        {
          ...fixtures.defaultData,
          border_radius: option,
        },
      );
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  aspect_ratio.enum.forEach(async option => {
    test(`aspect_ratio items: ${option}`, async () => {
      const results = await render(
        '@bolt-components-video-thumbnail/video-thumbnail.twig',
        {
          ...fixtures.defaultData,
          aspect_ratio: option,
        },
      );
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  test(`video_title_display`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        video_title_display: true,
        video: {
          title: 'Video Title Display',
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`button_attributes`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        button_attributes: {
          'data-test-attributes': 'Test Data',
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`button_attributes`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        button_attributes: {
          'data-test-attributes': 'Test Data',
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
  test(`video.duration`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        video: {
          title: 'Video Title Button Attributes',
          duration: '3:55',
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`video.has_subtitles`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        video: {
          title: 'Video Title Subtitles',
          has_subtitles: true,
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`video.content`, async () => {
    const results = await render(
      '@bolt-components-video-thumbnail/video-thumbnail.twig',
      {
        ...fixtures.defaultData,
        video: {
          content: fixtures.videoObject,
          title: 'Video Title Inline Play',
        },
      },
    );
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});
