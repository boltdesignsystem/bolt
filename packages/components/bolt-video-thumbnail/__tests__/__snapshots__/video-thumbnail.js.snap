// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Video Thumbnail Component Prop - aspect_ratio items: 1:1 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-1:1">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - aspect_ratio items: 4:3 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-4:3">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - aspect_ratio items: 16:9 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - border_radius items: large 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-large c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - border_radius items: none 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - border_radius items: small 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-small c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - button_attributes 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <button data-test-attributes="Test Data"
          class="c-bolt-video-thumbnail__inline-button"
          type="button"
          aria-label="Play the video"
  >
    <div class="c-bolt-video-thumbnail__content">
      /fixtures/1200x660.jpg
    </div>
    <div class="c-bolt-video-thumbnail__play-icon"
         aria-hidden="true"
    >
    </div>
  </button>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - static display 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - video.content 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9 c-bolt-video-thumbnail--inline">
  <button class="js-bolt-video-thumbnail-inline-button c-bolt-video-thumbnail__inline-button"
          type="button"
          aria-label="Play the video Video Title Inline Play"
  >
    <div class="c-bolt-video-thumbnail__content">
      /fixtures/1200x660.jpg
    </div>
    <div class="c-bolt-video-thumbnail__play-icon"
         aria-hidden="true"
    >
    </div>
  </button>
  <div class="c-bolt-video-thumbnail__content-video js-bolt-video-thumbnail-content-video">
    <video-js data-account="1900410236"
              data-player="O3FkeBiaDz"
              data-embed="default"
              data-video-id="4892122320001"
              controls
              class="c-base-video"
    >
    </video-js>
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - video.duration 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
  <div class="c-bolt-video-thumbnail__meta c-bolt-video-thumbnail__meta--bottom">
    <div class="c-bolt-video-thumbnail__duration">
      <span class="u-bolt-visuallyhidden">
        Video duration:
      </span>
      3:55
    </div>
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - video.has_subtitles 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
  <div class="c-bolt-video-thumbnail__meta c-bolt-video-thumbnail__meta--bottom">
    <div class="c-bolt-video-thumbnail__subtitles"
         aria-label="Subtitles available"
    >
      <svg class="e-bolt-icon e-bolt-icon--small"
           xmlns="http://www.w3.org/2000/svg"
           viewbox="0 0 32 32"
           aria-hidden="true"
      >
        <path fill="#151619"
              d="M29.8 5.2H2.2C1 5.2 0 6.3 0 7.4v17.2c0 1.2 1 2.2 2.2 2.2h27.6c1.2 0 2.2-1 2.2-2.2V7.4c0-1.1-1-2.2-2.2-2.2zM5.7 11.4H9c.8 0 1.4.6 1.4 1.4s-.5 1.3-1.4 1.3H5.7c-.7 0-1.3-.6-1.3-1.4s.6-1.3 1.3-1.3zM17.9 21H5.8c-.8 0-1.4-.6-1.4-1.4 0-.7.6-1.3 1.3-1.3h12.1c.8 0 1.4.6 1.5 1.3-.1.8-.6 1.4-1.4 1.4zm8.4.1H23c-.8 0-1.4-.6-1.4-1.4s.6-1.4 1.4-1.4h3.4c.7 0 1.3.6 1.4 1.4-.1.8-.7 1.4-1.5 1.4zm0-7.2h-12c-.6 0-1.2-.6-1.4-1.4 0-.8.6-1.4 1.4-1.4h12.1c.7 0 1.3.6 1.4 1.4-.1.8-.7 1.4-1.5 1.4z"
        >
        </path>
      </svg>
      <span class="u-bolt-visuallyhidden">
        Subtitles available
      </span>
    </div>
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component Prop - video.show_title 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__meta c-bolt-video-thumbnail__meta--top">
    <span class="c-bolt-video-thumbnail__title">
      Video Title Display
    </span>
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;

exports[`Bolt Video Thumbnail Component default 1`] = `
<div class="c-bolt-video-thumbnail c-bolt-video-thumbnail--border-radius-none c-bolt-video-thumbnail--aspect-ratio-16:9">
  <div class="c-bolt-video-thumbnail__content">
    /fixtures/1200x660.jpg
  </div>
  <div class="c-bolt-video-thumbnail__play-icon"
       aria-hidden="true"
  >
  </div>
</div>
`;
